modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -486,4 +488,26 @@ public static boolean isCompileTime(ExtensionModel extensionModel) {
     return extensionModel.getModelProperty(CompileTimeModelProperty.class).isPresent();
   }
 
+  /**
+   * Indicates whether a give {@link EnrichableModel} is synthetic or not.
+   *
+   * @param enrichableModel the enrichable model that may be synthetic
+   * @return if the enrichable model is synthetic or not
+   * @since 4.3.0
+   */
+  public static boolean isSynthetic(EnrichableModel enrichableModel) {
+    return enrichableModel.getModelProperty(SyntheticModelModelProperty.class).isPresent();
+  }
+
+  /**
+   * Tests the given {@code connectionProviderModel} for a {@link ImplementingTypeModelProperty} and if present it
+   * returns the enclosed implementing type.
+   *
+   * @param connectionProviderModel the provider to get the implemented type from
+   * @return an {@link Optional} that either has the provider implementing type, or is empty
+   * @since 4.3.0
+   */
+  public static Optional<Class> getImplementingType(ConnectionProviderModel connectionProviderModel) {
```

Comment:
This should receive EnrichableModel and should be used not only for Connection providers. Look for usages of ImplementingTypeModelProperty and change it everywhere (boyscout rule)